mypy/config_parser.py

```diff
@@ -124,6 +124,7 @@ def check_follow_imports(choice: str) -> str:
     'cache_dir': expand_path,
     'python_executable': expand_path,
     'strict': bool,
+    'exclude': lambda s: [p.strip() for p in s.split('\n') if p.strip()],
```

Comment:
Possible root cause for #11825?

Reply:
Yes - seems likely. Good find. The config_parser uses very basic newlines to split apart different excludes.

Reply:
The behavior is documented in the `config_file.rst` - about multiple patterns in a newline separated manner.

Didn't realize that multiline excludes like this were an old pattern
Suggestion on what to do to fix?

Reply:
`config_parser.py` is new to me, and I'm still trying to digest it. At a high level, I'm wondering if one can treat `.toml` files differently? It looks like there's a mechanism to do that, but I'm still trying to figure out if it's sufficient for what I have in mind, i.e., in a `.toml` file, if it's a string, just parse it as a single regex, but if it's an array, treat each item as its own regex? (Just thinking out loud here.)

Reply:
Looking further, I _**think**_ this might be as simple as appending the following [here](https://github.com/nipunn1313/mypy/blob/3f43c830f37489045ef8fd8e21d232aef42b28ef/mypy/config_parser.py#L132-L146):

``` python
toml_config_types.update({
    # …
    'exclude': try_split,
})
```

Dunno what testing looks like yet, though….

Reply:
Yep - that makes sense to me.
Something like
```
exclude: lambda exc: exc.strip() if isinstance(exc, str) else [e.strip() for e in exc]
```
?
Then exclude in a toml can be explicitly an array (instead of newline separated).

Reply:
A test could go in here https://github.com/python/mypy/blob/master/test-data/unit/cmdline.pyproject.test - similar to the cmdline.test one added in this diff.

Reply:
Okay, how about something like this?

``` python
def str_or_seq_of_strs(v: Union[str, Sequence[str]]) -> List[str]:
    if isinstance(v, str):
        return [v.strip()] if v.strip() else []
    return [p.strip() for p in v if p.strip()]

toml_config_types.update({
    # …
    'exclude': str_or_seq_of_strs,
})
```

Reply:
If I know you, you're probably ¾ done with a PR by now, but if you haven't started, I can take a stab….

Reply:
I haven't started. Go for it!
Appreciate the compliment :)

Reply:
I haven't looked at the details, but there's also this PR that was opened recently: https://github.com/python/mypy/pull/11746

Reply:
Thanks for the heads-up @hauntsaninja! I don't think that one works. I'll comment there.

Reply:
#11828 is up. I commented on #11746.